torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -176,6 +177,38 @@ def __iter__(self) -> Iterator[Tuple[str, io.IOBase]]:
             yield path, decompressor(file)
 
 
+class RarArchiveReader(IterDataPipe[Tuple[str, io.BufferedIOBase]]):
```

Comment:
This should be landed in `torchdata`. See pytorch/data#41.

Reply:
it's not for now. should I wait until it is?

Reply:
Nope, let's move forward here. If the PR in `torchdata` is merged, we can remove it here.